examples/usercmodule/cexample/examplemodule.c

```diff
@@ -0,0 +1,19 @@
+#include "py/obj.h"
```

Comment:
IMO it'd make sense to just cut-and-paste the C example from the docs, so it includes comments about each part of the source (rather than have a different example here compared to the docs, which may confuse readers when they expect to find the docs example code in this directory)

Reply:
Sure. Questions:
- example starts with 'include required definitions' but only obj.h is really required, can I just drop the others? Or do they serve to indicate what the public API is, in which case the comment would better state that as such?
- name 'cexample' can be kept instead of 'example' (just to match with 'cppexample')?
- add similar comments into the C++ example?

Reply:
> example starts with 'include required definitions' but only obj.h is really required, can I just drop the others? Or do they serve to indicate what the public API is, in which case the comment would better state that as such?

True.  The only one that's needed is runtime.h (it's a superset of obj.h b/c it includes obj.h, and adds some useful things like mp_raise etc).  And best practice is to just include runtime.h (and then anything else on an as-needed basis).  So my suggestion would be to just have `runtime.h` and remove the other two from the docs so they match the code example.

> name 'cexample' can be kept instead of 'example' (just to match with 'cppexample')?

Yes.

> add similar comments into the C++ example?

If you have time, would be good.

Reply:
Ok, ended up rewriting a large part of the docs because just stripping the code and saying 'look in examples' left it quite incomplete and confusing, imo. @andrewleech you wrote the original, any comments? So basically:
- move code from the docs to actual source code, mostly unchanged apart from the include file
- have the docs refer to the source code
- move the part which talks about the PP define under the 'Compiling the cmodule into MicroPython' section to make it clear 2 build actions are required
- add examples both for building the example modules, and a typical user project directory